core/admin/mailu/sso/views/base.py

```diff
@@ -48,6 +49,81 @@ def login():
             flask.flash('Wrong e-mail or password', 'error')
     return flask.render_template('login.html', form=form, fields=fields)
 
+@sso.route('/proxy', methods=['GET'])
+def proxy():
+    client_ip = flask.request.headers.get('X-Real-IP', flask.request.remote_addr)
+    username=flask.request.headers.get('X-Auth-Email') or ""
+    secret=flask.request.headers.get('X-Auth-Proxy-Secret') or ""
+    target=flask.request.headers.get('X-Auth-Proxy-Target') or ""
+    tokens=username.split("@")
+    domain_name=""
+    if (len(tokens)==2):
+        domain_name=tokens[1]
+        localpart=tokens[0]
+    domain = models.Domain.query.get(domain_name)
+    if not domain or domain=="None":
+        return flask.redirect(flask.url_for('.login'))
+
+    form = forms.LoginForm()
+    form.submitAdmin.label.text = form.submitAdmin.label.text + ' Admin'
+    form.submitWebmail.label.text = form.submitWebmail.label.text + ' Webmail'
+
+    fields = []
+    if str(app.config["WEBMAIL"]).upper() != "NONE":
+        fields.append(form.submitWebmail)
+    if str(app.config["ADMIN"]).upper() != "FALSE":
+        fields.append(form.submitAdmin)
+    fields = [fields]
+
+    if str(target).upper()=="ADMIN":
+        destination = "0; url="+ app.config['WEB_ADMIN']
+    else:
+        destination = "0; url="+ app.config['WEB_WEBMAIL']
+
+    payload_dict = {
+        'Refresh': destination
+    }
+
+    device_cookie, device_cookie_username = utils.limiter.parse_device_cookie(flask.request.cookies.get('rate_limit'))
+    if username != device_cookie_username and utils.limiter.should_rate_limit_ip(client_ip):
+        flask.flash('Too many attempts from your IP (rate-limit)', 'error')
+        return flask.render_template('login.html', form=form, fields=fields)
+    if utils.limiter.should_rate_limit_user(username, client_ip, device_cookie, device_cookie_username):
+        flask.flash('Too many attempts for this user (rate-limit)', 'error')
+        return flask.render_template('login.html', form=form, fields=fields)
+
+    if not (app.config['PROXY_SECRET'] in (None, '') or not app.config['PROXY_SECRET']) and app.config['PROXY_SECRET'] == secret and username != "":
```

Comment:
This is too complicated :)

What about:

```
if username and len(secret)>XX and app.config.get('PROXY_SECRET', "") == secret:
```

Reply:
Adopted ;)